GH-123299: Copyedit 3.14 What's New: Build Changes (#138488)
python/cpython: Doc/whatsnew/3.14.rst

  (Removed in :gh:`133079`, see also :gh:`130396`.)

Build Changes
=============

* GNU Autoconf 2.72 is now required to generate :file:`configure`.
  (Contributed by Erlend Aasland in :gh:`115765`.)

* ``wasm32-unknown-emscripten`` is now a :pep:`11` tier 3 platform.
  (Contributed by R. Hood Chatham in :gh:`127146`, :gh:`127683`, and :gh:`136931`.)

* ``#pragma``-based linking with ``python3*.lib`` can now be switched off
  with :c:expr:`Py_NO_LINK_LIB`.
  (Contributed by Jean-Christophe Fillion-Robin in :gh:`82909`.)

> * CPython now enables defaults a set of recommended compiler options

Reviewer: ```suggestion
* CPython now enables a set of recommended compiler options by default
```

  See :ref:`whatsnew314-tail-call` for further details.

* To disable the new remote debugging support, use the
  :option:`--without-remote-debug` :file:`configure` option.
  This may be useful for security reasons.

.. _whatsnew314-build_details:

:file:`build-details.json`
--------------------------

Installations of Python now contain a new file, :file:`build-details.json`.
This is a static JSON document containing build details for CPython,
to allow for introspection without needing to run code.
This is helpful for use-cases such as Python launchers, cross-compilation,
and so on.

:file:`build-details.json` must be installed in the platform-independent
standard library directory. This corresponds to the :ref:`'stdlib'
<installation_paths>` :mod:`sysconfig` installation path,
which can be found by running ``sysconfig.get_path('stdlib')``.

.. seealso::
   :pep:`739` -- ``build-details.json`` 1.0 -- a static description file
   for Python build details

.. _whatsnew314-no-more-pgp:

Discontinuation of PGP signatures
---------------------------------

PGP (Pretty Good Privacy) signatures will not be provided
for releases of Python 3.14 or future versions.

> To verify CPython artefacts, users must use `Sigstore verification materials

Reviewer: According to other parts of the document, the American English spelling "artifacts" is preferred.

```suggestion
To verify CPython artifacts, users must use `Sigstore verification materials
```

Author: Either spelling is fine, I think.

Reviewer 2: Let's stick with the original "artifacts" spelling, that's what the linked [PEP 761](https://peps.python.org/pep-0761/). https://www.python.org/downloads/metadata/sigstore/ and https://www.sigstore.dev use, and is also the current spelling used elsewhere in the docs. There's no "artefact" in the docs.

Author: Done

<https://www.python.org/downloads/metadata/sigstore/>`__.

> Releases have been signed using Sigstore since Python 3.11.
> 
> This change in release process was specified in :pep:`761`.
> 
> .. _Sigstore: https://www.sigstore.dev/

Reviewer 2: Is this unused?

```suggestion
Releases have been signed using Sigstore since Python 3.11.

This change in release process was specified in :pep:`761`.
```

Or was it meant to be used here?
```suggestion
Releases have been signed using Sigstore_ since Python 3.11.

This change in release process was specified in :pep:`761`.

.. _Sigstore: https://www.sigstore.dev/
```

Author: Oops, yes, thanks for spotting. Too many rounds of rephrasing!
